data/events/scripts/player.lua

```diff
@@ -741,14 +741,26 @@ function Player:onGainSkillTries(skill, tries)
 	end
 
 	-- Event scheduler skill rate
-	local STAGES_DEFAULT = skillsStages or nil
-	local SKILL_DEFAULT = self:getSkillLevel(skill)
-	local RATE_DEFAULT = configManager.getNumber(configKeys.RATE_SKILL)
+	if configManager.getBoolean(configKeys.RATE_USE_STAGES) then
+		STAGES_DEFAULT = skillsStages
+		SKILL_DEFAULT = self:getSkillLevel(skill)
+		RATE_DEFAULT = configManager.getNumber(configKeys.RATE_SKILL)
+	else 
+		STAGES_DEFAULT = nil
+		SKILL_DEFAULT = self:getSkillLevel(skill)
+		RATE_DEFAULT = configManager.getNumber(configKeys.RATE_SKILL)
+	end
```

Comment:
This way you avoid code repetition:

```suggestion
	-- Event scheduler skill rate
	if configManager.getBoolean(configKeys.RATE_USE_STAGES) then
		STAGES_DEFAULT = skillsStages
	else
		STAGES_DEFAULT = nil
	end
	SKILL_DEFAULT = self:getSkillLevel(skill)
	RATE_DEFAULT = configManager.getNumber(configKeys.RATE_SKILL)
```